### PR TITLE
feat(property): add property name

### PR DESCRIPTION
**Pull request by XuNeo**
### Description of the feature or fix

Binding language like `lua` or `js` prefer to use name to set property.
This patch is one of the series to make bindings easier to set widget property.

Added a table to convert property name to ID. The name table is generated automatically though `properties.py` script, based on the property ID definitions.

If the name is not provided for specified widget, then recursively looking up its base classes. `bsearch` is used to match up strings.

The test has passed.

This API is experimental and is subject to change if binding language requires differently.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/widgets/property/lv_style_properties.c [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+#include "../../core/lv_obj_property.h"
+
+#if LV_USE_OBJ_PROPERTY && LV_USE_OBJ_PROPERTY_NAME
+
+
+/**
+ * Style property names, name must be in order.
+ * Generated code from properties.py
+ */
+const lv_property_name_t style_property_names[111] = {
```
Comment (FASTSHIFT) [inline src/widgets/property/lv_style_properties.c]: lv_style_property_names

File: scripts/properties.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env python3
```
Comment (FASTSHIFT) [inline scripts/properties.py:1]: How do we prevent users from forgetting to call properties.py to update this file after adding a style?

Comment (XuNeo) [reply scripts/properties.py:1]: I'm planning to add property.py to CI similar to lv_conf_generate.py. Let's firstly review if the proposed method is the right way.

**Review by W-Mai — APPROVED** [review]

**Comment by W-Mai** [comment]
LGTM

File: docs/others/obj_property.rst [hunk 3]
```diff
@@ -35,6 +37,8 @@ Two APIs are provided to get/set widget properties. It can be enabled by setting
     lv_result_t lv_obj_set_property(lv_obj_t * obj, const lv_property_t * value);
     lv_property_t lv_obj_get_property(lv_obj_t * obj, lv_prop_id_t id);
 
+    lv_prop_id_t lv_obj_property_name2id(const lv_obj_class_t * clz, const char * name);
```
Comment (kisvegabor) [inline docs/others/obj_property.rst]:
```suggestion
    lv_prop_id_t lv_obj_get_property_id(const lv_obj_class_t * clz, const char * name);
```

**Comment by kisvegabor** [comment]
It seems to be very useful for us with the editor too.  It looks good to me, but @coevin @liamHowatt and @richardgazdik could you also take look at it?

File: docs/others/obj_property.rst (hunk 3 above)
Comment (XuNeo) [reply docs/others/obj_property.rst]: I recall that micropython has a naming constraint that the first parameter for lv_obj_xxx APIs should be `lv_obj_t *`.
Is it still true?

`lv_obj_get_` sounds like to get something from an object. But here it intends to convert a property name to ID based on its class. How about `lv_obj_property_get_id`?

File: src/widgets/property/lv_obj_property_names.h [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+#define LV_OBJ_PROPERTY_NAMES_H
+
+#include "../../misc/lv_types.h"
+
+#if LV_USE_OBJ_PROPERTY && LV_USE_OBJ_PROPERTY_NAME
+
+    extern const lv_property_name_t lv_style_property_names[111];
+    extern const lv_property_name_t lv_image_property_names[11];
+    extern const lv_property_name_t lv_obj_property_names[48];
+    /* *INDENT-ON* */
```
Comment (XuNeo) [inline src/widgets/property/lv_obj_property_names.h]: Remove.

**Review by coevin — APPROVED** [review]
Looks good to me!

File: docs/others/obj_property.rst (hunk 3 above)
Comment (kisvegabor) [reply docs/others/obj_property.rst]: I've just copied the name from the [code](https://github.com/lvgl/lvgl/pull/6329/files#diff-c9ad3a7d72566d008dd5214f92aae68ce51e57b0aa2de961cc373e448698f9aaR174). :slightly_smiling_face: 

But you are right, the `lv_obj_property` prefix seems more appropriate.

File: src/core/lv_obj_property.c [hunk 5]
```diff
@@ -165,23 +207,37 @@ static lv_result_t obj_property(lv_obj_t * obj, lv_prop_id_t id, lv_property_t *
...
+                    }
+                case LV_PROPERTY_TYPE_COLOR: {
+                        if(set)((lv_property_set_color_t)prop->setter)(obj, value->color);
+                        else value->color = ((lv_property_get_color_t)(prop->getter))(obj);
+                        break;
+                    }
+                case LV_PROPERTY_TYPE_POINT: {
+                        lv_point_t * point = &value->point;
+                        if(set)((lv_property_set_pointer_t)(prop->setter))(obj, point);
+                        else *point = ((lv_property_get_point_t)(prop->getter))(obj);
```
Comment (liamHowatt) [inline src/core/lv_obj_property.c]: Should `lv_property_set_pointer_t` be `lv_property_set_point_t`? You declared `lv_property_set_point_t` at the top.

**Review by liamHowatt — COMMENTED** [review]
Cool metaprogramming and use of bsearch :slightly_smiling_face:

File: src/widgets/property/lv_obj_properties.c [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+ * Obj widget property names, name must be in order.
+ * Generated code from properties.py
+ */
+/* *INDENT-OFF* */
+const lv_property_name_t lv_obj_property_names[48] = {
+    {"flag_adv_hittest",       LV_PROPERTY_OBJ_FLAG_ADV_HITTEST,},
+    {"flag_checkable",         LV_PROPERTY_OBJ_FLAG_CHECKABLE,},
+    {"flag_click_focusable",   LV_PROPERTY_OBJ_FLAG_CLICK_FOCUSABLE,},
+    {"flag_clickable",         LV_PROPERTY_OBJ_FLAG_CLICKABLE,},
+    {"flag_end",               LV_PROPERTY_OBJ_FLAG_END,},
```
Comment (kisvegabor) [inline src/widgets/property/lv_obj_properties.c:22]: Can we avoid adding it?

Comment (XuNeo) [reply src/widgets/property/lv_obj_properties.c:22]: Are you concerned about the flash size? `flag` is treated as a property so it's automatically included here. Maybe we should remove the `flag` property totally?

**Review by liamHowatt — APPROVED** [review]

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**